llmfoundry/models/layers/ffn.py

```diff
@@ -76,6 +74,7 @@ def build_ffn(
             device=device,
         )
     elif ffn_type == 'te_ln_mlp':
+        assert te is not None
```

Comment:
This should be an error return to the user right? e.g please install transformer engine..

Reply:
The error is actually in the config validation (https://github.com/mosaicml/llm-foundry/blob/bd8127252c660e45ed01413645d29427f86c085a/llmfoundry/models/mpt/configuration_mpt.py#L218-L227), so at this point we can assume the import has already been checked and this assert is really just for pyright.